src/app/components/payment-dialog/payment-dialog.component.ts

```diff
@@ -48,19 +49,31 @@ export class PaymentDialogComponent implements OnInit {
         this.form = this.fb.group({
             target_address: [data.targetAddress, Validators.required],
             amount: ['', Validators.required],
-            token: [data.tokenAddress, Validators.required],
+            token: [data.token, Validators.required],
             payment_identifier: '',
         });
     }
 
-    ngOnInit() {}
+    ngOnInit() {
+        this.form.controls.token.valueChanges
+            .pipe(takeUntil(this.ngUnsubscribe))
```

Comment:
Nitpick: this `takeUntil(subject)` is a little bit weird, if the subject is created only for cleanup. A more streamlined way would be to save subscriptions, and then `.add` multiple if needed and `.unsubscribe()` at the end. If you do that because you need `complete` callbacks for cleanup, check the `finalize` operator.

Reply:
We had some problem in tests with the old `add()` - `unsubscribe()` pattern. I don't remember the exact reason, but was something about the mocked observables. That's why I moved to the `takeUntil()` approach. Seems also to be recommended for Angular: https://stackoverflow.com/questions/38008334/angular-rxjs-when-should-i-unsubscribe-from-subscription